packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1536,6 +1576,8 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
     ): void {
         this.verifyNotClosed();
 
+        assert(this.context.pendingLocalState === undefined, "op submitted while processing pending initial state");
```

Comment:
Can you please check the stack when this assert fires? Will it reach some handler that closes container?
We might want to be more explicit here and just close container